go/src/koding/db/mongodb/modelhelper/machine.go

```diff
@@ -154,6 +154,19 @@ func GetMachinesByUsername(username string) ([]*models.Machine, error) {
 	return findMachine(query)
 }
 
+func GetAllMachinesByUsername(username string) ([]*models.Machine, error) {
```

Comment:
I second. Ideally we should have only one `Get` per model which accepts some `Filter` struct. This way if more filtering is needed, one just add one more field to the filter.

```go
f := &MachineFilter{
    Participated: true,
    Username:     "ppknap",
}

m, err := GetMachines(f)
```

Mocking such API in test would then be way easier. Just a thought.